pkg/validation/cluster.go

```diff
@@ -335,6 +335,20 @@ func ValidateClusterNetworkConfig(n *kubermaticv1.ClusterNetworkingConfig, cni *
 			fmt.Sprintf("%s proxy mode can be used only when Konnectivity is enabled", resources.EBPFProxyMode)))
 	}
 
+	if n.IPFamily == kubermaticv1.IPFamilyDualStack && dc != nil {
```

Comment:
cc @PratikDeoghare now I realised that with this we won't be able to easily test dual-stack on not yet supported providers (as it won't be allowed by webhook). What do you think about this? I think it is good to have this check here for production, but would be good to have some escape path for testing...

Reply:
If we could read these providers from a configmap then we could patch it for tests?

Reply:
yeah, or should we do it in `KubermaticConfiguration`?

Reply:
That should work even better.

Reply:
Still not 100% sure if we should complicate this with KC.
Maybe the webhook should check only the providers with datacenter-specific IPv6 support (for now OpenStack, next would be vsphere) and allow dual-stack for all other providers (for now).

Reply:
agreed :)